Unity-Technologies/ScriptableRenderPipeline
# Use real types for compute shadow sampling

### Purpose of this PR
Allow shadow sampling using "real" typed data.  Previous attempts either left shadow sampling incompatible (mixing half and float interfaces in the plumbing between core and HDRP) - this is the current state of master, or broken for mobile platforms.

---
### Testing status
**Katana Tests**: First off we need to make sure the Katana SRP tests are green?

**Manual Tests**: What did you do?
- [ x] Opened test project + Run graphic tests locally
- [ x] Built a player
- [ x] C# and shader warnings (supress shader cache to see them)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

Launch Katana (Link on master here - update for your branch):
https://katana.bf.unity3d.com/projects/com.unity.render-pipelines/builders?ScriptableRenderLoop_branch=master&automation-tools_branch=master&unity_branch=trunk&sort=1-asc

Alternative, launch Yamato (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/78-ScriptableRenderPipeline

Any test projects to go with this to help reviewers?

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?
Low
**Halo Effect**: None, Low, Medium, High?
Low - Desktop should be unaffected, and Universal does not use the compute sampling routines.
---
### Comments to reviewers
Notes for the reviewers you have assigned.
